src/main/java/teammates/ui/controller/AdminActivityLogPageAction.java

```diff
@@ -70,31 +75,138 @@ protected ActionResult execute() throws EntityDoesNotExistException{
+        if (data.isPersonSpecified()) {
+            String targetUserGoogleId = data.getPersonSpecified();
+            targetTimeZone = getLocalTimeZoneForRequest(targetUserGoogleId, "");
+        }
+        
+        if (targetTimeZone == Const.DOUBLE_UNINITIALIZED) { // if no person is specified or the person doesn't really exist
+            if ((logs.size() >= RELEVANT_LOGS_PER_PAGE) && (earliestLogChecked != null)) {
+                String userGoogleId = earliestLogChecked.getGoogleId();
+                String userRole = earliestLogChecked.getRole();
+                targetTimeZone = this.getLocalTimeZoneInfo(userGoogleId, userRole);
+            } else {
+                targetTimeZone = Const.SystemParams.ADMIN_TIMZE_ZONE_DOUBLE;
+            }
+        }
+        
+        double adminTimeZone = Const.SystemParams.ADMIN_TIMZE_ZONE_DOUBLE;
+        String timeInAdminTimeZone = computeLocalTime(adminTimeZone, String.valueOf(earliestSearchTime));
+        String timeInUserTimeZone =  computeLocalTime(targetTimeZone, String.valueOf(earliestSearchTime));
+        status += "The earliest log entry checked on <b>" + timeInAdminTimeZone + "</b> in Admin Time Zone (" 
+                  + adminTimeZone + ") and ";
+        if (targetTimeZone != Const.DOUBLE_UNINITIALIZED) {
+            status += "on <b>" + timeInUserTimeZone + "</b> in Local Time Zone (" + targetTimeZone + ").<br>";
+        } else {
+            status += timeInUserTimeZone;
+        }
+        
+        // the "Search More" button to continue searching from the previous fromDate 
+        status += "<button class=\"btn-link\" id=\"button_older\" onclick=\"submitFormAjax(" + data.getFromDate() + ");\">Search More</button>";
+        
+        status += "<input id=\"ifShowAll\" type=\"hidden\" value=\""+ data.getIfShowAll() +"\"/>";
+        status += "<input id=\"ifShowTestData\" type=\"hidden\" value=\""+ data.getIfShowTestData() +"\"/>";
+        
+        data.setStatusForAjax(status);
+        statusToUser.add(new StatusMessage(status, StatusMessageColor.INFO));
+    }
+
+    private List<ActivityLogEntry> searchLogsWithTimeIncrement(AdminActivityLogPageData data) {
+        List<ActivityLogEntry> appLogs = new LinkedList<ActivityLogEntry>();
+        
+        int numberOfSearchTimes = 0;
+        totalLogsSearched = 0;
+        while ((numberOfSearchTimes < MAX_SEARCH_TIMES) && (appLogs.size() < RELEVANT_LOGS_PER_PAGE)) {
+            // set fromDate is two-hour away from toDate
+            long nextTwoHour = data.getToDate() - SEARCH_TIME_INCREMENT;
+            data.setFromDate(nextTwoHour);
+            numberOfSearchTimes++;
+            
+            LogQuery query = buildQuery(data);
+            List<ActivityLogEntry> searchResult = searchLogsByQuery(query, data);
+            if (!searchResult.isEmpty()) {
+                appLogs.addAll(searchResult);
+            }
+            data.setToDate(data.getFromDate());
```

Comment:
Same question as above: do you think there will be a duplicate if there was a log at precisely the fromDate of the previous query?